### ISSUE TITLE
Fix S1481: do not report on variables used inside lambdas

### ISSUE THREAD

**Issue by fmallet** [SonarSource/sonar-csharp#316]
The following code reproduces the problem:

```
using System;
using System.Collections.Generic;
using System.Linq;

namespace SonarQubeConsoleApplication
{
    public class S1481
    {
        public IEnumerable<Type> Get(Type type) => 
            Enumerable.Empty<Type>().Where(o =>
                {
                    var a = typeof(object);
                    return o.Equals(a);
                });
    }
}
```

**Comment by michalb-sonar** [comment]
This code does not raise the issue, it was presumably fixed in the meantime.

It is reasonable not to require all lambda parameters to be used, but if there are unused local variables inside, the rule should report that. This is exactly what happens currently. I've added more tests to document this behaviour: https://github.com/SonarSource/sonar-csharp/pull/1040
